### ISSUE TITLE
python NetApp-Lib error

### ISSUE THREAD

**Issue by autocoder10** [ansible-collections/netapp#71]
##### SUMMARY
Getting below error when executing the playbook even though NetApp-Lib is installed. 
fatal: [localhost]: FAILED! => {"changed": false, "msg": "the python NetApp-Lib module is required"}

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
na_ontap_info

##### ANSIBLE VERSION
```
# ansible --version
ansible 2.9.13
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Sep 26 2019, 13:23:47) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
[root@rwle020v ~]# 

```

##### CONFIGURATION
```
# ansible-config dump --only-changed
DISPLAY_SKIPPED_HOSTS(/etc/ansible/ansible.cfg) = False
```
##### OS / ENVIRONMENT
NAME="Red Hat Enterprise Linux Server"
VERSION="7.8 (Maipo)"

##### STEPS TO REPRODUCE
I have installed requests, python netapp-lib version 2020.7.16 and ran source command on "collections-env-setup" file but still receiving the error "fatal: [localhost]: FAILED! => {"changed": false, "msg": "the python NetApp-Lib module is required"}"

Name: netapp-lib
Version: 2020.7.16
Summary: netapp-lib is required for Ansible deployments to interact with NetApp storage systems.
Home-page: https://netapp.io/
Author: NetApp, Inc.
Author-email: ng-ansibleteam@netapp.com
License: Apache License, Version 2.0
Location: /usr/local/lib/python3.6/site-packages
Requires: xmltodict, lxml

$ echo $PYTHONPATH
/home/schowd1/.ansible/collections

```
---
- hosts: localhost
  gather_facts: false
  collections:
    - netapp.ontap

  environment:
    SSL_CERT_FILE: <cert_path>

  vars:
    ansible_python_interpreter: /usr/bin/python3
    vserver: xxxxxxxx
    login: &login
      hostname: xxxxxxxx
      username: admin
      password: admins_password
      https: True
      validate_certs: True

  tasks:
  - name: Info
    na_ontap_info:
      state: info
      vserver: "{{ vserver }}"
      gather_subset: clock_info
      <<: *login
    register: ontap

  - name: print info
    debug:
      var: ontap
```

##### EXPECTED RESULTS
Need to show the NTP clock info for the vserver.

##### ACTUAL RESULTS
```
$ ansible-playbook test_gatherinfo.yml -vvv
ansible-playbook 2.9.13
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/test/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Sep 26 2019, 13:23:47) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
Using /etc/ansible/ansible.cfg as config file
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Parsed /etc/ansible/hosts inventory source with ini plugin

PLAYBOOK: test_gatherinfo.yml ***************************************************************************************************************************************************************
1 plays in test_gatherinfo.yml

PLAY [localhost] ****************************************************************************************************************************************************************************
META: ran handlers
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: test
<127.0.0.1> EXEC /bin/sh -c 'echo ~test && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/test/.ansible/tmp `"&& mkdir "` echo /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486 `" && echo ansible-tmp-1605239571.93-26137-244036756566486="` echo /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486 `" ) && sleep 0'
Using module file /home/test/.ansible/collections/ansible_collections/netapp/ontap/plugins/modules/na_ontap_info.py
<127.0.0.1> PUT /home/test/.ansible/tmp/ansible-local-261274tHnob/tmpv3Gacv TO /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486/AnsiballZ_na_ontap_info.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486/ /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486/AnsiballZ_na_ontap_info.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'SSL_CERT_FILE=<cert_path> /usr/bin/python3 /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486/AnsiballZ_na_ontap_info.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/test/.ansible/tmp/ansible-tmp-1605239571.93-26137-244036756566486/ > /dev/null 2>&1 && sleep 0'

TASK [Info] *********************************************************************************************************************************************************************************
task path: /home/test/netappplaybooks/test_gatherinfo.yml:21
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "cert_filepath": null, 
            "continue_on_error": [
                "never"
            ], 
            "desired_attributes": null, 
            "feature_flags": {}, 
            "gather_subset": [
                "clock_info"
            ], 
            "hostname": "xxxxxxx", 
            "http_port": null, 
            "https": true, 
            "key_filepath": null, 
            "max_records": 1024, 
            "ontapi": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "query": null, 
            "state": "info", 
            "summary": false, 
            "use_native_zapi_tags": false, 
            "use_rest": "auto", 
            "username": "admin", 
            "validate_certs": true, 
            "volume_move_target_aggr_info": null, 
            "vserver": "xxxxxxxx"
        }
    }, 
    "msg": "the python NetApp-Lib module is required"
}

PLAY RECAP **********************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

```

**Comment by lonico** [comment]
>> Location: /usr/local/lib/python3.6/site-packages

You installed netapp-lib for python3.6, but you're using 

>> python version = 2.7.5 

when running Ansible.

Make sure the versions are matching.

**Comment by lonico** [comment]
BTW, we have a slack channel that may provide faster answers.   And is much more interactive.

Look for the slack link on https://netapp.io

**Comment by autocoder10** [comment]
Correct but in the playbook I added the "ansible_python_interpreter" in vars to use python3. Also, Initially I got the xmltodict error which is for python3 and after adding the below line I am not seeing the error. So, if it is able to check and confirm xmltodict but why it's not able detect "netapp-lib" which is in the same location.

  **vars:
    ansible_python_interpreter: /usr/bin/python3**

Name: xmltodict
Version: 0.12.0
Summary: Makes working with XML feel like you are working with JSON
Home-page: https://github.com/martinblech/xmltodict
Author: Martin Blech
Author-email: martinblech@gmail.com
License: MIT
**Location: /usr/local/lib/python3.6/site-packages**
Requires:

**Comment by lonico** [comment]
$ ansible-playbook test_gatherinfo.yml -vvv
ansible-playbook 2.9.13
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/test/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Sep 26 2019, 13:23:47) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]


This shows 2.7.5.

**Comment by autocoder10** [comment]
Right.. that's the default python that comes with Ansible... But when I add this "ansible_python_interpreter: /usr/bin/python3" in the playbook it will point to python3.

**Comment by lonico** [comment]
Yes, I can see /usr/bin/python3 in the trace.

But I wonder if it's looking for /usr/lib/python3.6/site-packages and not /usr/local/lib/python3.6/site-packages.

You could try to start /usr/bin/python3 and run
>>> import sys
>>> sys.path

In my installation, I only see /usr/lib and my home directory.

**Comment by autocoder10** [comment]
From python3 I'm able to import zapi from netapp-lib which is in "/usr/local/lib/python3.6/site-packages". Also, I did import sys and it's showing all the paths for python3.

# python3
Python 3.6.8 (default, Sep 26 2019, 11:57:09) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from netapp_lib.api.zapi import zapi
>>> 

>>> import sys
>>> sys.path
['', '/usr/lib64/python36.zip', '/usr/lib64/python3.6', '/usr/lib64/python3.6/lib-dynload', '/usr/local/lib64/python3.6/site-packages', '**/usr/local/lib/python3.6/site-packages'**, '/usr/lib64/python3.6/site-packages', '/usr/lib/python3.6/site-packages']
>>> 

# pip3 show netapp-lib
Name: netapp-lib
Version: 2020.7.16
Summary: netapp-lib is required for Ansible deployments to interact with NetApp storage systems.
Home-page: https://netapp.io/
Author: NetApp, Inc.
Author-email: ng-ansibleteam@netapp.com
License: Apache License, Version 2.0
Location: /usr/local/lib/python3.6/site-packages
Requires: xmltodict, lxml

**Comment by lonico** [comment]
On my system, I have two installations of python3.  A bit surprising, but they are different:

```
% which python3
/usr/local/bin/python3

% /usr/bin/python3
Python 3.8.2 (default, Sep 24 2020, 19:37:08) 
[Clang 12.0.0 (clang-1200.0.32.21)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'requests'
>>> ^D

% /usr/local/bin/python3
Python 3.6.2 (default, Sep 22 2017, 19:58:31) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> 
```

**Comment by lonico** [comment]
We created a new module to help debug such issues:
https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/ontap/plugins/modules/na_ontap_debug.py

This does not truly fix the issue, but would help understand environment issues.

Addressed with 0e641bf61a57bb52d7dea4167eed1ac55b13a30f

**Comment by pescobar** [comment]
I was having the same problem. The error was happening in [these lines](https://github.com/ansible-collections/netapp/blob/master/ansible_collections/netapp/ontap/plugins/module_utils/netapp.py#L54-L56):

```
try:
    from netapp_lib.api.zapi import zapi
    HAS_NETAPP_LIB = True
```

when executing those lines in my python interpreter I was getting this error:

```
>>> from netapp_lib.api.zapi import zapi                                                                                                                                                                                               
Traceback (most recent call last):                                                                                                                                                                                                     
    import six                                                                                                                                                                                                                         
ModuleNotFoundError: No module named 'six'  
```

I fixed it by reinstalling the missing library with `pip install six`
